### ISSUE TITLE
warn Package @ant-design/react-native contains invalid configuration: "dependency.assets" is not allowed. Please verify it's properly linked using "react-native config" command and contact the package maintainers about this. error: unknown command 'link'

### ISSUE THREAD

**Issue by douguohai** [ant-design/ant-design-mobile-rn#1289]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/antd-reproduction-template-forked-jyh2k9?file=/package.json)

### Steps to reproduce

初始化项目报错

### What is expected?

期望修复这个错误

### What is actually happening?

期望修复这个错误

| Environment | Info |
| --- | --- |
| antd | 5.0.4 |
| React | 0.71.3 |
| System | MACOS |
| Browser | chrome |

**Comment by eliomaria** [comment]
Try this:

React Native 0.69 update
In 0.69 version of React Native link command has been removed. react-native-asset  should be used to automatically link the font assets. Just run the following command:
$ npx react-native-asset


Other thing you can try to refactor the react-native.config.js inside the @ant-design package in node modules for:
The one inside react-native:
module.exports = {
  project: {
    ios: {},
    android: {
      sourceDir: './rn-kitchen-sink/android'
    }
  },
  assets: ['../icons-react-native/fonts']
};

and the one inside icons-react-native:
module.exports = {
  project: {
    ios: {},
    android: {}
  },
  assets: ['./src/assets/fonts']
};

**Comment by Oakyden** [comment]
@eliomaria can you elaborate a bit more?

I am in the root of my project and run 'npx react-native-asset' and it says `Error: Cannot find module '/Users/**username**/Documents/Repositories/**projectName**/react-native.config.js'`

I presume i need to cd into the node modules folder? Or provide some kind of flag on the command?

Also surely editing the files you state above in node modules is not sustainable, so we need to create a patch file or similar?

**Comment by avrahm** [comment]
@Oakyden 

I was running into the same wanrings but also with @ant-design/icons-react-native. 

I cleared the warning messages and resolved the link using a combination of steps from both issues. But I do agree updating node_modules is not sustainable. 

```
warn Package @ant-design/icons-react-native contains invalid configuration: "dependency.assets" is not allowed. 
Please verify it's properly linked using "react-native config" command and contact the package maintainers about this.

warn Package @ant-design/react-native contains invalid configuration: "dependency.assets" is not allowed. 
Please verify it's properly linked using "react-native config" command and contact the package maintainers about this.
```

### First run the solution by @huzhanbo1996 in https://github.com/ant-design/ant-design-icons/issues/535#issuecomment-1604027589

**NOTE:** I updated the steps for `npm` and clarified some other parts.
>  
> 1. npm I react-native-asset
> 2. Create or update the root config file react-native.config.js like this
> 
> ```js
> module.exports = {
>    assets: ["node_modules/@ant-design/icons-react-native/fonts"]
> };
> ```
> 
> 4. Add this into app.json
> 
> ```json
> "packagerOpts": {
>     "config": "metro.config.js"
>   }
> ```
> 
> 5. npx react-native-asset  -> this will copy assets into platform specific path


### Next, I complete the steps from @eliomaria in https://github.com/ant-design/ant-design-mobile-rn/issues/1289#issuecomment-1457878465

**NOTE:** I updated the steps to clarify some points and fixed the asset line for step 3.

> 6. npx react-native-asset (can be skipped if you just ran from step 5 above)
> 
> 7. Refactor the **dependency** listed in react-native.config.js inside the @ant-design package in node modules. 
>
> a. This is what you will find:
>  
>   ```js
>   dependency: {
>     assets: ['../icons-react-native/fonts'],
>   },
>  ```
>  
> b. Replace it with:
>  
> ```js
> module.exports = {
>   project: {
>     ios: {},
>     android: {
>       sourceDir: './rn-kitchen-sink/android'
>     }
>   },
>   assets: ['../icons-react-native/fonts']
> };
> ```
> 
> 8. Update the **dependency** inside the node_modules for icons-react-native:
> 
> ```js
> module.exports = {
>   project: {
>     ios: {},
>     android: {}
>   },
>   assets: ['fonts']
> };
> ```

Best of luck with your project!

**Comment by Oakyden** [comment]
Thanks @avrahm, I've since moved on to using React Paper, but I hope your reply helps someone!

**Comment by buihuynhngoctram** [comment]
> @Oakyden
> 
> I was running into the same wanrings but also with @ant-design/icons-react-native.
> 
> I cleared the warning messages and resolved the link using a combination of steps from both issues. But I do agree updating node_modules is not sustainable.
> 
> ```
> warn Package @ant-design/icons-react-native contains invalid configuration: "dependency.assets" is not allowed. 
> Please verify it's properly linked using "react-native config" command and contact the package maintainers about this.
> 
> warn Package @ant-design/react-native contains invalid configuration: "dependency.assets" is not allowed. 
> Please verify it's properly linked using "react-native config" command and contact the package maintainers about this.
> ```
> 
> ### First run the solution by @huzhanbo1996 in [ant-design/ant-design-icons#535 (comment)](https://github.com/ant-design/ant-design-icons/issues/535#issuecomment-1604027589)
> **NOTE:** I updated the steps for `npm` and clarified some other parts.
> 
> > 1. npm I react-native-asset
> > 2. Create or update the root config file react-native.config.js like this
> > 
> > ```js
> > module.exports = {
> >    assets: ["node_modules/@ant-design/icons-react-native/fonts"]
> > };
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 4. Add this into app.json
> > 
> > ```json
> > "packagerOpts": {
> >     "config": "metro.config.js"
> >   }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 5. npx react-native-asset  -> this will copy assets into platform specific path
> 
> ### Next, I complete the steps from @eliomaria in [#1289 (comment)](https://github.com/ant-design/ant-design-mobile-rn/issues/1289#issuecomment-1457878465)
> **NOTE:** I updated the steps to clarify some points and fixed the asset line for step 3.
> 
> > 6. npx react-native-asset (can be skipped if you just ran from step 5 above)
> > 7. Refactor the **dependency** listed in react-native.config.js inside the @ant-design package in node modules.
> > 
> > a. This is what you will find:
> > ```js
> > dependency: {
> >   assets: ['../icons-react-native/fonts'],
> > },
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > b. Replace it with:
> > ```js
> > module.exports = {
> >   project: {
> >     ios: {},
> >     android: {
> >       sourceDir: './rn-kitchen-sink/android'
> >     }
> >   },
> >   assets: ['../icons-react-native/fonts']
> > };
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 8. Update the **dependency** inside the node_modules for icons-react-native:
> > 
> > ```js
> > module.exports = {
> >   project: {
> >     ios: {},
> >     android: {}
> >   },
> >   assets: ['fonts']
> > };
> > ```
> 
> Best of luck with your project!

Thanks a lot, it works for me 👍

**Comment by yydddiiii** [comment]
> @Oakyden
> 
> I was running into the same wanrings but also with @ant-design/icons-react-native.
> 
> I cleared the warning messages and resolved the link using a combination of steps from both issues. But I do agree updating node_modules is not sustainable.
> 
> ```
> warn Package @ant-design/icons-react-native contains invalid configuration: "dependency.assets" is not allowed. 
> Please verify it's properly linked using "react-native config" command and contact the package maintainers about this.
> 
> warn Package @ant-design/react-native contains invalid configuration: "dependency.assets" is not allowed. 
> Please verify it's properly linked using "react-native config" command and contact the package maintainers about this.
> ```
> 
> ### First run the solution by @huzhanbo1996 in [ant-design/ant-design-icons#535 (comment)](https://github.com/ant-design/ant-design-icons/issues/535#issuecomment-1604027589)
> **NOTE:** I updated the steps for `npm` and clarified some other parts.
> 
> > 1. npm I react-native-asset
> > 2. Create or update the root config file react-native.config.js like this
> > 
> > ```js
> > module.exports = {
> >    assets: ["node_modules/@ant-design/icons-react-native/fonts"]
> > };
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 4. Add this into app.json
> > 
> > ```json
> > "packagerOpts": {
> >     "config": "metro.config.js"
> >   }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 5. npx react-native-asset  -> this will copy assets into platform specific path
> 
> ### Next, I complete the steps from @eliomaria in [#1289 (comment)](https://github.com/ant-design/ant-design-mobile-rn/issues/1289#issuecomment-1457878465)
> **NOTE:** I updated the steps to clarify some points and fixed the asset line for step 3.
> 
> > 6. npx react-native-asset (can be skipped if you just ran from step 5 above)
> > 7. Refactor the **dependency** listed in react-native.config.js inside the @ant-design package in node modules.
> > 
> > a. This is what you will find:
> > ```js
> > dependency: {
> >   assets: ['../icons-react-native/fonts'],
> > },
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > b. Replace it with:
> > ```js
> > module.exports = {
> >   project: {
> >     ios: {},
> >     android: {
> >       sourceDir: './rn-kitchen-sink/android'
> >     }
> >   },
> >   assets: ['../icons-react-native/fonts']
> > };
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 8. Update the **dependency** inside the node_modules for icons-react-native:
> > 
> > ```js
> > module.exports = {
> >   project: {
> >     ios: {},
> >     android: {}
> >   },
> >   assets: ['fonts']
> > };
> > ```
> 
> Best of luck with your project!

非常感谢！！我用这个方法也解决了问题！！

**Comment by 1uokun** [comment]
fixed in 5.1.2

see the detail in https://github.com/ant-design/ant-design-mobile-rn/blob/master/docs/react/introduce.en-US.md#2-installation

**Comment by Chenjiujiu** [comment]
> 已在 5.1.2 中修复
> 
> 中的详细信息 [请参阅https://github.com/ant-design/ant-design-mobile-rn/blob/master/docs/react/introduce.en-US.md#2-installation ](https://github.com/ant-design/ant-design-mobile-rn/blob/master/docs/react/introduce.en-US.md#2-installation)

It  exists in version 5.4.2
